pkg/koordlet/resourceexecutor/resctrl.go

```diff
@@ -0,0 +1,147 @@
+/*
+Copyright 2022 The Koordinator Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resourceexecutor
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"k8s.io/klog/v2"
+
+	"github.com/koordinator-sh/koordinator/pkg/koordlet/util/system"
+)
+
+const ErrResctrlDir = "resctrl path or file not exist"
+const CacheIdIndex = 2
+
+func NewResctrlReader() ResctrlReader {
+	if vendorId, err := system.GetVendorIDByCPUInfo(system.GetCPUInfoPath()); err != nil {
+		// FIXME: should we panic there?
+		klog.V(0).ErrorS(err, "get cpu vendor error")
```

Comment:
Please notice many virtual machines may not define a known vendor. For the scenarios, the QoS plugin and collector of the Resctrl QoS should abort the resctrl setting since the VM environment is unsupported. In this function, perhaps we should throw the error and let the caller abort. It is common where a Kubernetes cluster manages both bare-metal and VM nodes, and the koordlet should not panic for this to fail other resource management functions.